dkpro-jwpl-wikimachine/src/main/java/org/dkpro/jwpl/wikimachine/decompression/BZip2Decompressor.java

```diff
@@ -35,19 +37,17 @@ public class BZip2Decompressor
     @Override
     public InputStream getInputStream(String fileName) throws IOException
     {
-        InputStream outputStream;
-
-        BufferedInputStream inputStream = new BufferedInputStream(new FileInputStream(fileName));
-        /*
-         * skip 2 first bytes (see the documentation of CBZip2InputStream) e.g. here
-         * http://lucene.apache.org/tika/xref/org/apache/tika/parser /pkg/bzip2
-         * /CBZip2InputStream.html
-         */
-        inputStream.skip(2);
-        outputStream = new BZip2CompressorInputStream(inputStream);
-
-        return outputStream;
-
+        final InputStream in;
+        if (fileName.contains(File.separator)) {
```

Comment:
The `getResourceAsStream` doesn't explicitly state, what is considered as a valid `name` of a resource. If we look into `getResources` the [JavaDoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ClassLoader.html#getResource(java.lang.String)) states:

> The name of a resource is a '/'-separated path name that identifies the resource.

`File.separator` will return `/` on a *unix-oid system, so checking by `/` to decide if file system orr classloader won't work in all cases.

Maybe we can introduce an additional variable to indicate whether to check via classloader or handle it as a file-system resource?

Reply:
How about using an URL?